corehq/apps/app_manager/models.py

```diff
@@ -5759,18 +5759,17 @@ class Meta(object):
 
     @classmethod
     def by_app(cls, app):
-        model, created = cls.objects.get_or_create(app_id=app.master_id, domain=app.domain, defaults={
-            'apk_version': LATEST_APK_VALUE,
-            'app_version': LATEST_APP_VALUE,
-        })
+        model = cls.by_app_id(app.domain, app.master_id)
         model._app = app
         return model
 
     @classmethod
     def by_app_id(cls, domain, app_id):
-        app = get_app(domain, app_id, latest=True, target='release')
-        assert app_id == app.master_id, "this class doesn't handle copy app ids"
-        return cls.by_app(app)
+        model, created = cls.objects.get_or_create(app_id=app_id, domain=domain, defaults={
```

Comment:
Hey @snopoke,
So now we are going to just create the record, without checking if the app id actually belongs to the master app?

And plus, now we don't raise the Assertion Error so it won't get caught in the view like earlier since we have moved the check to later but fail when calling `get_latest_app_version` which will raise an uncaught assertion error?

Reply:
good point about not catching the assertion in the view

Yes, this will create an object in the DB even it's not a 'master' app ID but that's not very likely. I don't think we've ever seen that in the wild (search sentry for "Received an invalid heartbeat request").

Reply:
I think its fine if we want to just create it now, but would be nice to just delete it if that is the case when fetching the app?

And how do we avoid fetching the app in the request now? looks like we are just delaying it?

Reply:
catch assertions: 9754f17d318

Reply:
> I think its fine if we want to just create it now, but would be nice to just delete it if that is the case when fetching the app?

I don't think this is necessary and it could create a lot of churn in that DB table which is more harmful

> And how do we avoid fetching the app in the request now? looks like we are just delaying it?

We only fetch it if there isn't a cache for the values.

Reply:
Oh cool, so the cache `get_latest_app_version` returns that without needing us to load app,
thanks!